server-common/src/main/java/org/apache/kafka/server/purgatory/DelayedOperation.java

```diff
@@ -29,10 +28,9 @@
  * a delayed fetch operation could be waiting for a given number of bytes to accumulate.
  * <br/>
  * The logic upon completing a delayed operation is defined in onComplete() and will be called exactly once.
- * Once an operation is completed, isCompleted() will return true. onComplete() can be triggered by either
- * forceComplete(), which forces calling onComplete() after delayMs if the operation is not yet completed,
- * or tryComplete(), which first checks if the operation can be completed or not now, and if yes calls
- * forceComplete().
+ * Once an operation is completed, isCompleted() will return true. onComplete() is called from forceComplete(),
+ * which is triggered once by either expiration if the operation is not completed after delayMs, or tryComplete()
```

Comment:
Actually, it's possible for forceComplete() to be called once from tryComplete() and another from expiration. So, we need to remove `once`.